### ISSUE TITLE
Can't uninstall: Command shutdown.exe /a exited with '1116'

### ISSUE THREAD

**Issue by kobeyu** [DarwinJS/ChocoPackages#13]
The uninstall script is failing for me. Whilst trying to run `choco uninstall openssh -params '"/DeleteConfigAndServerKeys"'`

I get:
```
PS C:\> choco uninstall openssh -params '"/DeleteConfigAndServerKeys"'
Chocolatey v0.10.2
Uninstalling the following packages:
openssh

openssh v0.0.1.0
The package openssh wants to run 'chocolateyuninstall.ps1'.
Note: If you don't run this script, the installation will fail.
Note: To confirm automatically next time, use '-y' or consider setting
 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.
Do you want to run the script?([Y]es/[N]o/[P]rint): y

Running on: Windows Web Server 2008 R2, (ServerWeb)
/DeleteConfigAndServerKeys - Removing SSH Config and Server Keys.
WARNING: Removing all config and server keys as requested by /DeleteConfigAndServerKeys

No rules match the specified criteria.

openssh uninstall not successful.
Error while running 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1'.
 See log for details.
openssh not uninstalled. An error occurred during uninstall:
 openssh uninstall not successful.

Chocolatey uninstalled 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - openssh (exited 1) - Error while running 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1'.
 See log for details.

If a package uninstall is failing and/or you've already uninstalled the
 software outside of Chocolatey, you can attempt to run the command
 with `-n` to skip running a chocolateyUninstall script, additionally
 adding `--skip-autouninstaller` to skip an attempt to automatically
 remove system-installed software. This will only remove the packaging
 files and not things like software installed to Programs and Features.

If a package is failing because it is a dependency of another package
 or packages, then you may first need to consider if it needs removed
 as it is typically installed as a dependency for a reason. If you
 decide that you still want to remove it, head into
 `$env:ChocolateyInstall\lib` and find the package folder you want
 removed. Then delete the folder for the package. This option should
 only be used as a last resort.


You use Chocolatey? You are amazing! Are you ready to take the next
 step and look even smarter with more awesome features?
 https://chocolatey.org/compare

```

chocolatey.log:
```
2016-10-05 13:59:36,154 [INFO ] - The package openssh wants to run 'chocolateyuninstall.ps1'.
2016-10-05 13:59:36,155 [INFO ] - Note: If you don't run this script, the installation will fail.
2016-10-05 13:59:36,157 [INFO ] - Note: To confirm automatically next time, use '-y' or consider setting 
 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.
2016-10-05 13:59:36,185 [INFO ] - Do you want to run the script?
2016-10-05 13:59:36,185 [INFO ] -  1) yes
2016-10-05 13:59:36,186 [INFO ] -  2) no
2016-10-05 13:59:36,187 [INFO ] -  3) print
2016-10-05 14:01:37,273 [INFO ] - Choice selected: 'yes'
2016-10-05 14:01:37,300 [DEBUG] - Calling built-in PowerShell host with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1' -installArguments '' -packageParameters '/DeleteConfigAndServerKeys'']
2016-10-05 14:01:37,303 [DEBUG] - Redirecting System.Management.Automation.resources, Version=3.0.0.0, Culture=en-US, PublicKeyToken=31bf3856ad364e35, requested by ''
2016-10-05 14:01:37,664 [DEBUG] - Host version is 4.0, PowerShell Version is '4.0' and CLR Version is '4.0.30319.42000'.
2016-10-05 14:01:37,861 [INFO ] - VERBOSE: Exporting function 'Format-FileSize'.
2016-10-05 14:01:37,861 [INFO ] - VERBOSE: Exporting function 'Get-ChecksumValid'.
2016-10-05 14:01:37,862 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyUnzip'.
2016-10-05 14:01:37,862 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyWebFile'.
2016-10-05 14:01:37,862 [INFO ] - VERBOSE: Exporting function 'Get-EnvironmentVariable'.
2016-10-05 14:01:37,862 [INFO ] - VERBOSE: Exporting function 'Get-EnvironmentVariableNames'.
2016-10-05 14:01:37,863 [INFO ] - VERBOSE: Exporting function 'Get-FtpFile'.
2016-10-05 14:01:37,863 [INFO ] - VERBOSE: Exporting function 'Get-OSArchitectureWidth'.
2016-10-05 14:01:37,863 [INFO ] - VERBOSE: Exporting function 'Get-ToolsLocation'.
2016-10-05 14:01:37,863 [INFO ] - VERBOSE: Exporting function 'Get-UACEnabled'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Get-UninstallRegistryKey'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Get-VirusCheckValid'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Get-WebFile'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Get-WebFileName'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Get-WebHeaders'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Install-BinFile'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyDesktopLink'.
2016-10-05 14:01:37,864 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyEnvironmentVariable'.
2016-10-05 14:01:37,865 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyExplorerMenuItem'.
2016-10-05 14:01:37,865 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyFileAssociation'.
2016-10-05 14:01:37,865 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyInstallPackage'.
2016-10-05 14:01:37,865 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPackage'.
2016-10-05 14:01:37,866 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPath'.
2016-10-05 14:01:37,866 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPinnedTaskBarItem'.
2016-10-05 14:01:37,866 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPowershellCommand'.
2016-10-05 14:01:37,866 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyShortcut'.
2016-10-05 14:01:37,867 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyVsixPackage'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyZipPackage'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Install-Vsix'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Set-EnvironmentVariable'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Set-PowerShellExitCode'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Start-ChocolateyProcessAsAdmin'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Test-ProcessAdminRights'.
2016-10-05 14:01:37,868 [INFO ] - VERBOSE: Exporting function 'Uninstall-BinFile'.
2016-10-05 14:01:37,869 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyEnvironmentVariable'.
2016-10-05 14:01:37,869 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyPackage'.
2016-10-05 14:01:37,869 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyZipPackage'.
2016-10-05 14:01:37,869 [INFO ] - VERBOSE: Exporting function 'Update-SessionEnvironment'.
2016-10-05 14:01:37,869 [INFO ] - VERBOSE: Exporting function 'Write-ChocolateyFailure'.
2016-10-05 14:01:37,870 [INFO ] - VERBOSE: Exporting function 'Write-ChocolateySuccess'.
2016-10-05 14:01:37,870 [INFO ] - VERBOSE: Exporting function 'Write-FileUpdateLog'.
2016-10-05 14:01:37,870 [INFO ] - VERBOSE: Exporting function 'Write-FunctionCallLogMessage'.
2016-10-05 14:01:37,871 [INFO ] - VERBOSE: Exporting alias 'Get-ProcessorBits'.
2016-10-05 14:01:37,872 [INFO ] - VERBOSE: Exporting alias 'Get-OSBitness'.
2016-10-05 14:01:37,872 [INFO ] - VERBOSE: Exporting alias 'Get-InstallRegistryKey'.
2016-10-05 14:01:37,873 [INFO ] - VERBOSE: Exporting alias 'Generate-BinFile'.
2016-10-05 14:01:37,873 [INFO ] - VERBOSE: Exporting alias 'Add-BinFile'.
2016-10-05 14:01:37,873 [INFO ] - VERBOSE: Exporting alias 'Start-ChocolateyProcess'.
2016-10-05 14:01:37,873 [INFO ] - VERBOSE: Exporting alias 'Invoke-ChocolateyProcess'.
2016-10-05 14:01:37,873 [INFO ] - VERBOSE: Exporting alias 'Remove-BinFile'.
2016-10-05 14:01:37,873 [INFO ] - VERBOSE: Exporting alias 'refreshenv'.
2016-10-05 14:01:37,874 [INFO ] - VERBOSE: Importing function 'Format-FileSize'.
2016-10-05 14:01:37,874 [INFO ] - VERBOSE: Importing function 'Get-ChecksumValid'.
2016-10-05 14:01:37,875 [INFO ] - VERBOSE: Importing function 'Get-ChocolateyUnzip'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-ChocolateyWebFile'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-EnvironmentVariable'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-EnvironmentVariableNames'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-FtpFile'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-OSArchitectureWidth'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-ToolsLocation'.
2016-10-05 14:01:37,876 [INFO ] - VERBOSE: Importing function 'Get-UACEnabled'.
2016-10-05 14:01:37,877 [INFO ] - VERBOSE: Importing function 'Get-UninstallRegistryKey'.
2016-10-05 14:01:37,877 [INFO ] - VERBOSE: Importing function 'Get-VirusCheckValid'.
2016-10-05 14:01:37,877 [INFO ] - VERBOSE: Importing function 'Get-WebFile'.
2016-10-05 14:01:37,877 [INFO ] - VERBOSE: Importing function 'Get-WebFileName'.
2016-10-05 14:01:37,877 [INFO ] - VERBOSE: Importing function 'Get-WebHeaders'.
2016-10-05 14:01:37,878 [INFO ] - VERBOSE: Importing function 'Install-BinFile'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyDesktopLink'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyEnvironmentVariable'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyExplorerMenuItem'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyFileAssociation'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyInstallPackage'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPackage'.
2016-10-05 14:01:37,879 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPath'.
2016-10-05 14:01:37,880 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPinnedTaskBarItem'.
2016-10-05 14:01:37,880 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPowershellCommand'.
2016-10-05 14:01:37,881 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyShortcut'.
2016-10-05 14:01:37,881 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyVsixPackage'.
2016-10-05 14:01:37,882 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyZipPackage'.
2016-10-05 14:01:37,882 [INFO ] - VERBOSE: Importing function 'Install-Vsix'.
2016-10-05 14:01:37,882 [INFO ] - VERBOSE: Importing function 'Set-EnvironmentVariable'.
2016-10-05 14:01:37,882 [INFO ] - VERBOSE: Importing function 'Set-PowerShellExitCode'.
2016-10-05 14:01:37,882 [INFO ] - VERBOSE: Importing function 'Start-ChocolateyProcessAsAdmin'.
2016-10-05 14:01:37,882 [INFO ] - VERBOSE: Importing function 'Test-ProcessAdminRights'.
2016-10-05 14:01:37,883 [INFO ] - VERBOSE: Importing function 'Uninstall-BinFile'.
2016-10-05 14:01:37,883 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyEnvironmentVariable'.
2016-10-05 14:01:37,883 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyPackage'.
2016-10-05 14:01:37,883 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyZipPackage'.
2016-10-05 14:01:37,883 [INFO ] - VERBOSE: Importing function 'Update-SessionEnvironment'.
2016-10-05 14:01:37,883 [INFO ] - VERBOSE: Importing function 'Write-ChocolateyFailure'.
2016-10-05 14:01:37,884 [INFO ] - VERBOSE: Importing function 'Write-ChocolateySuccess'.
2016-10-05 14:01:37,884 [INFO ] - VERBOSE: Importing function 'Write-FileUpdateLog'.
2016-10-05 14:01:37,884 [INFO ] - VERBOSE: Importing function 'Write-FunctionCallLogMessage'.
2016-10-05 14:01:37,884 [INFO ] - VERBOSE: Importing alias 'Add-BinFile'.
2016-10-05 14:01:37,885 [INFO ] - VERBOSE: Importing alias 'Generate-BinFile'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'Get-InstallRegistryKey'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'Get-OSBitness'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'Get-ProcessorBits'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'Invoke-ChocolateyProcess'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'refreshenv'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'Remove-BinFile'.
2016-10-05 14:01:37,886 [INFO ] - VERBOSE: Importing alias 'Start-ChocolateyProcess'.
2016-10-05 14:01:37,936 [DEBUG] - Running 'ChocolateyScriptRunner' for openssh v0.0.1.0 with packageScript 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1', packageFolder:'C:\ProgramData\chocolatey\lib\openssh', installArguments: '', packageParameters: '/DeleteConfigAndServerKeys',
2016-10-05 14:01:37,946 [DEBUG] - Running 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1'
2016-10-05 14:01:37,982 [INFO ] - Running on: Windows Web Server 2008 R2, (ServerWeb)
2016-10-05 14:01:38,029 [INFO ] - /DeleteConfigAndServerKeys - Removing SSH Config and Server Keys.
2016-10-05 14:01:38,032 [INFO ] - 
2016-10-05 14:01:44,075 [WARN ] - WARNING: Removing all config and server keys as requested by /DeleteConfigAndServerKeys
2016-10-05 14:01:44,165 [INFO ] - 
2016-10-05 14:01:44,166 [INFO ] - No rules match the specified criteria.
2016-10-05 14:01:44,167 [INFO ] - 
2016-10-05 14:01:44,239 [DEBUG] - Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1' -installArguments '' -packageParameters '/DeleteConfigAndServerKeys''] exited with '1'.
2016-10-05 14:01:44,244 [DEBUG] - Calling command ['"C:\Windows\System32\shutdown.exe" /a']
2016-10-05 14:01:44,278 [DEBUG] - Command ['"C:\Windows\System32\shutdown.exe" /a'] exited with '1116'
2016-10-05 14:01:44,279 [ERROR] - openssh uninstall not successful.
2016-10-05 14:01:44,284 [ERROR] - Error while running 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1'.
 See log for details.
2016-10-05 14:01:44,286 [ERROR] - openssh not uninstalled. An error occurred during uninstall:
 openssh uninstall not successful.
2016-10-05 14:01:44,320 [WARN ] - 
Chocolatey uninstalled 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2016-10-05 14:01:44,322 [INFO ] - 
2016-10-05 14:01:44,322 [ERROR] - Failures
2016-10-05 14:01:44,326 [ERROR] -  - openssh (exited 1) - Error while running 'C:\ProgramData\chocolatey\lib\openssh\tools\chocolateyuninstall.ps1'.
 See log for details.
2016-10-05 14:01:44,328 [WARN ] - 
If a package uninstall is failing and/or you've already uninstalled the
 software outside of Chocolatey, you can attempt to run the command 
 with `-n` to skip running a chocolateyUninstall script, additionally
 adding `--skip-autouninstaller` to skip an attempt to automatically 
 remove system-installed software. This will only remove the packaging 
 files and not things like software installed to Programs and Features.

If a package is failing because it is a dependency of another package
 or packages, then you may first need to consider if it needs removed
 as it is typically installed as a dependency for a reason. If you 
 decide that you still want to remove it, head into 
 `$env:ChocolateyInstall\lib` and find the package folder you want 
 removed. Then delete the folder for the package. This option should
 only be used as a last resort.
 
2016-10-05 14:01:44,334 [WARN ] - 
You use Chocolatey? You are amazing! Are you ready to take the next 
 step and look even smarter with more awesome features?
 https://chocolatey.org/compare
2016-10-05 14:01:44,339 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2016-10-05 14:01:44,359 [DEBUG] - Exiting with 1
```

I get the same results on this windows 2008 server, and my local windows 10 pro desktop.

**Comment by kobeyu** [comment]
I should say, it does actually remove all of the files in C:\Program Files\OpenSSH-Win64, (but not the directory itself), but because chocolatey thinks the uninstall failed, chocolatey thinks it's still installed. I know I can fix this, but thought you'd like to know.

**Comment by DarwinJS** [comment]
Please gist such long log files.

I believe this was fixed in 0.0.0.901 and in the current 0.0.1.0.

The work around is to upgrade to the new version and then do an uninstall.

**Comment by kobeyu** [comment]
sorry about the log dump. I'll gist them next time. All of this installing and uninstalling was done today, with version 0.0.1.0. `choco upgrade openssh` says I've got the most recent one installed.

**Comment by DarwinJS** [comment]
This is fixed in the 0.0.2.0 chocolatey package - the entire folder is removed on uninstall.  Will have to re-add code back to preserve keys once I understand more about how to accomplish that.

Release Notes (from https://chocolatey.org/packages/openssh/0.0.2.0):
- NEW: /SSHAgentFeature - enables SSH agent for use with client tools when /SSHServerFeature is not used.
- NEW: Sample Dockerfile included for Server Core 2016 or Nano.
- FIX: Uninstall improved.
- NEW: InstallChoco_and_win32-openssh.ps1 and InstallChoco_and_win32-openssh_with_server.ps1 both refresh the environment after installing OpenSSH so it can be used in the same console it was installed in.
- FIX: ssh-add is only run after ssh-agent is started.
- FIX: sshd and ssh-agent are always started (if installed), however, a warning to reboot is still generated if key based authentication is being used.

Please reopen if you find problems.
